test/spanner.ts

```diff
@@ -4985,6 +4993,69 @@ describe('Spanner with mock server', () => {
       assert.deepStrictEqual(operations1, operations2);
     });
   });
+
+  it('Check for span annotations', () => {
+    const exporter = new InMemorySpanExporter();
+    const provider = new NodeTracerProvider({
+      sampler: new AlwaysOnSampler(),
+      exporter: exporter,
+    });
+    provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+    provider.register();
+
+    after(async () => {
+      await provider.shutdown();
+    });
+
+    startTrace('aSpan', {opts: {tracerProvider: provider}}, span => {
+      const database = newTestDatabase();
+
+      async function runIt() {
+        const query = {
+          sql: 'SELECT 1',
+        };
+
+        const [rows] = await database.run(query);
+        assert.strictEqual(rows.length, 1);
+      }
+
+      runIt();
+
+      span.end();
+
+      const spans = exporter.getFinishedSpans();
+      assert.strictEqual(spans.length, 1, 'Exactly 1 span');
+      const span0 = spans[0];
+      const events = span0.events;
+
+      // Sort the events by earliest time of occurence.
+      events.sort((evtA, evtB) => {
+        return evtA.time < evtB.time;
+      });
+
+      const gotEventNames: string[] = [];
+      events.forEach(event => {
+        gotEventNames.push(event.name);
+      });
+
+      const wantEventNames = [
+        'Requesting 25 sessions',
```

Comment:
These event names, doesn't look right? 

In case of success , shouldn't we have event like "Acquired session" and we shouldn't have 'Waiting for a session to become available' as it should do a cache hit.

Reply:
So with how fragile the code in this file is, the creation of the Spanner object is precarious and is tightly coupled with the assertions of the test invocations and checks for suffixes with values like `*-instance-3`, so I made a TODO at the top of this `it` to later on refactor this code so that we can create our own Spanner object and check for all the events below. Those mentioned flakes were a huge delay for this review.